### DISCUSSION TITLE
Как удалять или дестроить инициализированный Tool?

### DISCUSSION THREAD

**Discussion by MiSeKor** [finsight/QUIKSharp#355] [Q&A]
Как удалять или дестроить инициализированный Tool?
Вот сделал таблицу с несколькими Tool, надо мне удалить один, выделил нажал ДЕЛЕТ, он из таблици пропал, но по факту, он ЕСТЬ, его подписки работают, если там для теста сделать вывод сообщений они будут выводится, так же как при его визуальном присутствии.

**Comment by Pr0phet1c** [comment]
Если в рамках инициализации Tool, и/или после оной Вы подписались на какие-то события по этому инструменту, то очевидно, что надо отписаться от них прежде чем дестроить инструмент.
